### PR TITLE
refactor: springframework dependencies multi module

### PR DESCRIPTION
**Pull request by aureamunoz**
I would change the repo name to `quarkus-spring-api` as it aims to wrap or include the dependencies under `org.springframework` groupId.

This goes with [this commit](https://github.com/quarkusio/quarkus/commit/57e3d76eb389418b764e19fd067f61a5bacfa9aa)

WDYT?

### CODE REVIEW

**Comment by geoand** [comment]
Is this what was discussed with @cescoffier?

**Comment by aureamunoz** [comment]
yes

**Comment by geoand** [comment]
I'm fine with it if the @cescoffier and @gsmet are

File: pom.xml [hunk 1]
```diff
@@ -34,141 +34,138 @@
...
-                  </includes>
-                </filter>
-              </filters>
-            </configuration>
-          </execution>
-        </executions>
-      </plugin>
-    </plugins>
-  </build>
+<!--  <build>-->
```
Comment (cescoffier) [inline pom.xml]: Should be removed.

File: quarkus-spring-beans-api/pom.xml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                                        <include>org/springframework/beans/factory/support/AbstractBeanDefinition**</include>
+                                    </includes>
+                                </filter>
+                            </filters>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
```
Comment (cescoffier) [inline quarkus-spring-beans-api/pom.xml]: Can you check that the resulting artifact does not force resolving the original artifacts and its dependencies?

File: quarkus-spring-context-api/src/main/java/Dummy.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+/***
+ * Dummy class for javadoc
+ *
+ */
+
```
Comment (cescoffier) [inline quarkus-spring-context-api/src/main/java/Dummy.java]: superfluous empty line

File: quarkus-spring-core-api/src/main/java/Dummy.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+/***
+ * Dummy class for javadoc
+ *
+ */
+
```
Comment (cescoffier) [inline quarkus-spring-core-api/src/main/java/Dummy.java]: superfluous empty line.

File: quarkus-spring-beans-api/pom.xml (hunk 2 above)
Comment (aureamunoz) [reply quarkus-spring-beans-api/pom.xml]: > LGTM, I made a few comments.
> 
> Now the main questions are:
> 
>     * should these artifacts have dependencies between them (like core depending on bean or whatever, I don't know the original graph)
I don't think so, not sure to understand this question
> 
>     * we need to double-check we don't resolve the original transitive dependencies
I seems ok.

**Pull request merged**